### ISSUE TITLE
merge command not allowing reordering of args

### ISSUE THREAD

**Issue by pmolodo** [twpayne/chezmoi#1354]
## Describe the bug

The [docs for the merge command](https://www.chezmoi.io/docs/reference/#merge-target) state that you can use the `merge.args` config setting to change how the args are passed into the `merge.cmd`.  However, it looks like there is a bug in the way the args are handled - they don't seem to be template expanded, and the default args are always appended to them.

That is - the docs state that the default value of `merge.args` is `["{{ .Destination }}", "{{ .Source }}", "{{ .Target }}"]`.  That would imply that this `chezmoi.toml`:

```
[merge]
    command = "p4merge"
```

...should behave the same as this:

```
[merge]
    command = "p4merge"
    args = ["{{ .Destination }}", "{{ .Source }}", "{{ .Target }}"]
```

However, they don't.  If I use the 2nd example (with the explicit `merge.args`), and run:

```
chezmoi merge --debug "~/myfile"
```

...then I will see:

```
2021-08-18T14:21:44-07:00 DBG RunCmd args=["p4merge","{{ .Destination }}","{{ .Source }}","{{ .Target }}","C:/Users/pmolodowitch/myfile","C:/Users/pmolodowitch/.local/share/chezmoi/myfile","C:\\Users\\PMOLOD~1\\AppData\\Local\\Temp\\chezmoi-merge1804468260/myfile"] dir=C:/Users/pmolodowitch path="C:\\Program Files\\Perforce\\p4merge.exe"
```

So it looks like there are two problems:
- The templates aren't being properly replaced
- The default args are always appended to the user-supplied args

## To reproduce

As noted above - set your `chezmoi.toml` to include:

```
[merge]
    command = "p4merge"
    args = ["{{ .Destination }}", "{{ .Source }}", "{{ .Target }}"]
```

...and then run

```
chezmoi merge --debug "~/somefile"
```

## Expected behavior

With the given args above, there should be no change in behavior.  (And ideally, you could re-order them, and see that reflected properly).

## Output of command with the `--verbose` flag

`--verbose` gives no additional info - but see above for the relevant output from `--debug`

## Output of `chezmoi doctor`

```console
$ chezmoi doctor
RESULT   CHECK                MESSAGE
ok       version              v2.1.2, commit a3fb755a88315bbdecedba3d849324752989400d, built at 2021-07-14T20:07:22Z, built by goreleaser
ok       os-arch              windows/amd64
ok       config-file          C:/Users/pmolodowitch/.config/chezmoi/chezmoi.toml is a file
ok       source-dir           C:/Users/pmolodowitch/.local/share/chezmoi is a directory
ok       suspicious-entries   no suspicious entries found in C:/Users/pmolodowitch/.local/share/chezmoi
ok       dest-dir             C:/Users/pmolodowitch is a directory
ok       shell                found C:\Program Files\Git\usr\bin\bash.exe
ok       editor               found C:\Program Files\Microsoft VS Code\bin\code.cmd
ok       git-cli              found C:\Program Files\Git\mingw64\bin\git.exe, version 2.31.1
ok       merge-cli            found C:\Program Files\Perforce\p4merge.exe
ok       age-cli              age not found in $PATH
ok       gnupg-cli            found C:\Program Files\Git\usr\bin\gpg.exe, version 2.2.27
info     1password-cli        op not found in $PATH
info     bitwarden-cli        bw not found in $PATH
info     gopass-cli           gopass not found in $PATH
info     keepassxc-cli        keepassxc-cli not found in $PATH
info     keepassxc-db         not set
info     lastpass-cli         lpass not found in $PATH
info     pass-cli             pass not found in $PATH
info     vault-cli            vault not found in $PATH
info     secret-cli           not set
```

**Comment by twpayne** [comment]
There's an explicit [test for this](https://github.com/twpayne/chezmoi/blob/master/internal/cmd/testdata/scripts/merge_unix.txt#L27).

`chezmoi doctor` says you're using version 2.1.2 but https://github.com/twpayne/chezmoi/commit/dc6011bf7fb28260a1853f91bef27fa369573ceb only got released in [2.1.4](https://github.com/twpayne/chezmoi/releases/tag/v2.1.4).

**Comment by pmolodo** [comment]
Ah! Sorry, didn't realize that was such a new addition.  My mistake, works after updating, closing the issue!
